spec/system/find_local_council_spec.rb

```diff
@@ -0,0 +1,443 @@
+          visit "/find-local-council"
+          fill_in("postcode", with: "CH25 9BJ")
+          click_on("Find")
+        end
+
+        it "includes the select address text in the title element" do
+          expect(page).to have_title("Find your local council: #{I18n.t('formats.local_transaction.select_address').downcase} - GOV.UK", exact: true)
+        end
+
+        it "prompts you to choose your address" do
+          expect(page).to have_content("Select an address")
+        end
+
+        it "displays a dropdown select" do
+          expect(page).to have_css(".govuk-select")
+        end
+
+        it "contains a list of addresses mapped to authority slugs" do
+          expect(page).to have_content("House 1")
+          expect(page).to have_content("House 2")
+          expect(page).to have_content("House 3")
+          expect(page).to have_content("House 4")
+          expect(page).to have_content("House 5")
+          expect(page).to have_content("House 6")
+          expect(page).to have_content("House 7")
+        end
+      end
+
+      context "when no local council is found" do
+        before do
+          stub_locations_api_has_no_location("XM4 5HQ")
+          visit "/find-local-council"
+          fill_in("postcode", with: "XM4 5HQ")
+          click_on("Find")
+        end
+
+        it "remains on the find your local council page" do
+          expect(current_path).to eq("/find-local-council")
+          expect(page).to have_content("Find your local council")
+        end
+
+        it "shows an error message" do
+          expect(page).to have_content("We couldn't find a council for this postcode.")
+        end
+      end
+    end
+  end
+
+  context "when entering a specific local authority slug in the URL" do
+    context "with valid slug" do
+      before do
+        stub_local_links_manager_has_a_local_authority("islington")
+        visit "/find-local-council/islington"
+      end
+
+      it "includes the search result text in the page title" do
+        expect(page).to have_title("Find your local council: #{I18n.t('formats.local_transaction.search_result')} - GOV.UK", exact: true)
+      end
+
+      it "shows the local authority result" do
+        expect(page).to have_content("Your local authority is")
+        within(".unitary-result") do
+          expect(page).to have_content("Islington")
+          expect(page).to have_link("Go to Islington website", href: "http://islington.example.com", exact: true)
+        end
+      end
+    end
+
+    context "with invalid slug" do
+      before do
+        stub_local_links_manager_does_not_have_an_authority("hogwarts")
+        visit "/find-local-council/hogwarts"
+      end
+
+      it "shows an error message" do
+        expect(page.status_code).to eq(404)
+      end
+    end
+  end
+end
+
+def stub_local_links_manager_has_a_district_and_unitary_local_authority(district_slug, unitary_slug, district_snac: "00AG", unitary_snac: "00LC", local_custodian_code: nil)
```

Comment:
No, it's subtly different (this is mimicking the unlikely but possible situation in which we have a district but the parent is a unitary rather than county LA. There isn't a test stub for that and it's only relevant here, so... seems reasonable to add a new method)